CMSgov/bluebutton-web-server
# [BB2-2011] Update logging events for Application model's data_access_type and end_date fields

**JIRA Ticket:**
[BB2-2011](https://jira.cms.gov/browse/BB2-2011)

**User Story or Bug Summary:**

We want to be able to add metrics to our Quicksight dashboards.  In order to do so, we need to add data to our logging routines.  We will add the data in real time and during the nightly processes.  The reason to do this work is to tracking our API usage and troubleshooting any issues.

Data to log:
 * research app expirations
 * application type field 
 * data access grant expiration dates
 * operations around data access limits

 *AC:*
 * Add real time application logging where it makes sense to do so
 * Add to the nightly global state metrics logging
 * Update units tests to validate added logging
 * If applicable, update selenium tests to added logging

*QA:*
 * Perform unit tests to make sure additional fields show in logging
 * If applicable, run some selenium tests to make sure logging is happening
 * 

### What Does This PR Do?


1. Added auth_app_data_access_type and auth_app_end_date to logging events where the concerned application attributes are part of the logging record, e.g. auth_app_name, auth_app_id etc., this will make the application's fields related to limit data access feature available in the logging events - hence available for the down stream components to consume.
2. Added BB2 to BFD request headers: BlueButton-AuthAppDataAccessType, BlueButton-AuthAppEndDate such that the newly introduced application attributes become available to the BFD side
3. refactored local dev provisioned limit_data_access waffle switch to waffle flag
4. 

### What Should Reviewers Watch For?



If you're reviewing this PR, please check these things, in particular:

* TODO


### What Security Implications Does This PR Have?

Submitters should complete the following questionnaire:

* If the answer to any of the questions below is **Yes**, then here's a link to the associated Security Impact Assessment (SIA), security checklist, or other similar document in Confluence: N/A.
    * Does this PR add any new software dependencies? **No**.
    * Does this PR modify or invalidate any of our security controls? **No**.
    * Does this PR store or transmit data that was not stored or transmitted before? **No**.
* If the answer to any of the questions below is **Yes**, then please add StewGoin as a reviewer, and note that this PR should not be merged unless/until he also approves it.
    * Do you think this PR requires additional review of its security implications for other reasons? **No**.


### What Needs to Be Merged and Deployed Before this PR?



This PR cannot be either merged or deployed until the following pre-requisite changes have been fully deployed:

* CMSgov/some_repo#42
  
### Any Migrations?



  * [ ] There is a more complicated migration plan (downtime, etc) 


### Submitter Checklist



I have gone through and verified that...:

    1. It doesn't unnecessarily tie a bunch of disparate features, fixes, refactorings, etc. together.
    5. There isn't too much of a burden on reviewers.
    6. Any problems it causes have a small "blast radius".
    7. It'll be easier to rollback if that becomes necessary.
* [x] I have named this PR and its branch such that they'll be automatically be linked to the (most) relevant Jira issue, per: <https://confluence.atlassian.com/adminjiracloud/integrating-with-development-tools-776636216.html>.
    * At least two other engineers on this project, at least one of whom is a senior engineer or owns the relevant component(s) here.
    * Any relevant engineers on other projects (e.g. BFD, SLS, etc.).
    * Please review the standards every few months to ensure you're familiar with them.
